cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamWriterSizeManager.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.stream.service;
+
+import co.cask.cdap.data.stream.service.heartbeat.StreamWriterHeartbeat;
+import com.google.common.util.concurrent.Service;
+
+/**
+ * This interface manages the sizes of data written by one stream writer run in a {@link StreamHandler}.
+ * For each stream, It sends {@link StreamWriterHeartbeat}s at regular intervals to notify listeners
+ * of the updated size of the stream.
+ */
+public interface StreamWriterSizeManager extends Service {
```

Comment:
Why need to separate size manager and size collector? Seems like size manager doesn't do anything beside start/stop.